src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Logging/ProjectOutputWindowProjectLogger.cs

```diff
@@ -30,37 +29,19 @@ public ProjectOutputWindowProjectLogger(IProjectThreadingService threadingServic
             _outputWindowProvider = outputWindowProvider;
         }
 
-        public void WriteLine(string format, params object[] arguments)
-        {
-            if (_options.IsProjectOutputPaneEnabled)
-            {
-                // Only allocate if the pane is actually enabled
-                string text = string.Format(CultureInfo.CurrentCulture, format, arguments);
-
-                WriteLineCore(text);
-            }
-        }
-
         public void WriteLine(string text)
         {
-            if (_options.IsProjectOutputPaneEnabled)
-            {
-                WriteLineCore(text);
-            }
-        }
-
-        private void WriteLineCore(string text)
-        {
             // Extremely naive implementation of a Windows Pane logger - the assumption here is that text is rarely written,
             // so transitions to the UI thread are uncommon and are fire and forget. If we start writing to this a lot (such 
             // as via build), then we'll need to implement a better queueing mechanism.
             _threadingService.Fork(async () => {
+                if (await _options.GetIsProjectOutputPaneEnabledAsync())
```

Comment:
@davkean Sorry, I should have made it clear that I just hacked the logging to get it working and was going to spend more time on it once other things were dealt with. I thought it would be nice to use the window you're adding, but there are threading issues since the up to date check isn't running on the UI thread. I tried to hack a solution, but it's probably better that I just use the logger I'm given (which goes to the build output window).